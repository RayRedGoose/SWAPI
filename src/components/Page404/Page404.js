import React from 'react'
import './Page404.scss'
import obiWan from '../../images/obiWan.svg'
import hand from '../../images/hand (1).svg'

const Page404 = ( ) => {
  return (
    <section className="no-page">
      <img src={obiWan} alt="Obi Wan Kenobi"/>
      <div>
        <img src={hand} alt="hand"/>
      </div>
      <h1>404</h1>
      <h2>There aren't the droids you're looking for...</h2>
    </section>
  )
}

export default Page404
