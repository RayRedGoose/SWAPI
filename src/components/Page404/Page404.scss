.no-page {
  background-color: rgba(0,0,0,0.2);
  display: flex;
  align-items: flex-start;
  height: 100vh;
  overflow: hidden;
  position: relative;
  &>img {
    width: 500px;
  }
  div {
    border-radius: 50%;
    width: 350px;height: 350px;
    position: absolute;
    top: 50%; left: 28%;
    animation-delay: 5s;
    animation: round-move 7s linear infinite;
    img {
      width: 230px;
      margin: 0 -45%;
      transform: rotate(-45deg);
    }
  }
  h1 {
    color: rgba(255,255,255,0.2);
    font-size: 300px;
    position: absolute;
    top: -60px; right: 0;
  }
  h2 {
    color: #5A2C05;
    font-size: 64px;
    width: 50%;
    margin: 260px 0 0 120px;
    text-shadow: 2px 2px 48px #fff;
    position: relative;
    z-index: 20;
  }
}

@keyframes round-move {
  0% {
    transform: rotate(-10deg);
    opacity: 0;
  }
  5% {
    opacity: 1;
  }
  95% {
    opacity: 1;
  }
  100% {
    transform: rotate(60deg);
    opacity: 0;
  }
}
