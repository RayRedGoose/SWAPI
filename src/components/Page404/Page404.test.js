import React from 'react'
import { shallow } from 'enzyme'
import Page404 from './Page404'

describe("Page404", () => {

  it('should render Page404 with all data passed in correctly', () => {
    const wrapper = shallow(<Page404 />)

    expect(wrapper).toMatchSnapshot()
  })
})
