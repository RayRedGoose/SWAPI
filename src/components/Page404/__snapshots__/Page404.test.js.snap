// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page404 should render Page404 with all data passed in correctly 1`] = `
<section
  className="no-page"
>
  <img
    alt="Obi Wan Kenobi"
    src="obiWan.svg"
  />
  <div>
    <img
      alt="hand"
      src="hand.svg"
    />
  </div>
  <h1>
    404
  </h1>
  <h2>
    There aren't the droids you're looking for...
  </h2>
</section>
`;
